js/modules/tamagotchi.js

```diff
@@ -5,6 +5,52 @@ export default class Tamagotchi {
 		this.energy = { value: 10, importance: 2 };
 		this.fun = { value: 10, importance: 4 };
 		console.log('Tamagotchi initialized');
+
+		const energyValue = '.energy';
+		const hungerValue = '.hunger';
+		const funValue = '.fun';
+		const healthValue = '.health';
+
+		this.energyDecrease = setInterval(() => {
+			let newDecrementValue = 2;
+
+			if (this.fun.value <= 0) {
+				this.energy.value -= newDecrementValue;
+				if (this.energy.value <= 0) {
+					this.energy.value = 0;
+				}
+			} else {
+				this.energy.value -= 1;
+			}
+			if (this.energy.value <= 0) {
+				clearInterval(this.energyDecrease);
+			}
+			this.displayEnergy(energyValue);
+		}, 2000);
+		this.hungerDecrease = setInterval(() => {
+			this.hunger.value -= 1;
+			if (this.hunger.value <= 0) {
+				clearInterval(this.hungerDecrease);
+			}
+			this.displayHunger(hungerValue);
+		}, 1000);
+		this.funDecrease = setInterval(() => {
+			this.fun.value -= 1;
+			if (this.fun.value <= 0) {
+				clearInterval(this.funDecrease);
+			}
+			this.displayFun(funValue);
+		}, 1000);
+		this.healthDecrease = setInterval(() => {
+			if (this.energy.value <= 0 || this.hunger.value <= 0) {
+				this.health.value -= 1;
```

Comment:
**issue:** It looks like health decreasing logic works well, but the problem seems to be with displaying wrong value in that specific moment. I suggest you to investigate `this.displayHealth(healthValue);`, and what is being displayed there. In case of further issues please share this problem in our group chat - it seems to be an interesting issue to solve together:)